python/bindings/decoder.hpp

```diff
@@ -0,0 +1,32 @@
+#pragma once
+
+#include "bindings_core.hpp"
+#include "py_database.hpp"
+#include "message_db_singleton.hpp"
+#include "py_message_objects.hpp"
+
+#include "novatel_edie/decoders/oem/header_decoder.hpp"
+#include "novatel_edie/decoders/oem/message_decoder.hpp"
+
+namespace nb = nanobind;
+
+namespace novatel::edie::oem {
+
+class PyDecoder
+{
+  public:
+    PyMessageDatabase::Ptr database;
+    HeaderDecoder header_decoder;
+    MessageDecoder message_decoder;
+
+    PyDecoder()
+        : database(MessageDbSingleton::get()), header_decoder(HeaderDecoder(MessageDbSingleton::get())), message_decoder(MessageDbSingleton::get())
```

Comment:
```suggestion
        : PyDecoder(MessageDbSingleton::get())
```
Can simplify it a bit with a delegating constructor.